tests/test_ptycho.py

```diff
@@ -296,11 +307,59 @@ def test_consistent_lstsq_grad(self):
             },
         )
 
+    def test_consistent_lstsq_grad_variable_probe(self):
```

Comment:
Does the probe variation only work with ls gradient?

Reply:
Yes. Adding this feature to `cgrad` will require some thought and some refactoring, so I have not included it here.